### PR TITLE
Feature | View and Edit User Permissions

### PR DESCRIPTION
**Pull request by ChazUK**
I also updated the Exit button here

### CODE REVIEW

**Comment by ChazUK** [comment]
[(12) User Accounts - View user and edit - Front end](https://trello.com/c/UWYy8JTK/183-12-user-accounts-view-user-and-edit-front-end)

File: src/app/features/workplace/user-account-edit-permissions/user-account-edit-permissions.component.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    private router: Router,
+    private formBuilder: FormBuilder,
+    private breadcrumbService: BreadcrumbService,
+    private errorSummaryService: ErrorSummaryService,
+    private userService: UserService
+  ) {
+    this.user = this.route.snapshot.data.user;
+    this.establishment = this.route.parent.snapshot.data.establishment;
+
+    this.return = { url: ['/workplace', this.establishment.uid], fragment: 'user-accounts' };
```
Comment (tgreyuk) [inline src/app/features/workplace/user-account-edit-permissions/user-account-edit-permissions.component.ts]: You will be able to get here from '/dashboard' aswell?

File: src/app/features/workplace/user-account-edit-permissions/user-account-edit-permissions.component.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    this.form = this.formBuilder.group({
+      role: [this.user.role, Validators.required],
+      primary: this.user.isPrimary,
+    });
+  }
+
+  public onSubmit(payload: { action: string; save: boolean } = { action: 'continue', save: true }) {
+    if (!payload.save) {
+      return this.router.navigate(['/workplace', this.establishment.uid], { fragment: 'user-accounts' });
```
Comment (tgreyuk) [inline src/app/features/workplace/user-account-edit-permissions/user-account-edit-permissions.component.ts]: Should this be using this.return?

File: src/app/features/workplace/user-account-edit-permissions/user-account-edit-permissions.component.ts (hunk 1 above)
Comment (tgreyuk) [reply src/app/features/workplace/user-account-edit-permissions/user-account-edit-permissions.component.ts]: That is a question rather than a statement?

**Comment by ChazUK** [comment]
[(3) User Accounts - Change primary user](https://trello.com/c/gKAfsyyh/146-3-user-accounts-change-primary-user)

File: src/app/features/workplace/user-account-edit-permissions/user-account-edit-permissions.component.ts (hunk 1 above)
Comment (ChazUK) [reply src/app/features/workplace/user-account-edit-permissions/user-account-edit-permissions.component.ts]: Very true, I will update it to use your return observable.

File: src/app/features/workplace/user-account-edit-permissions/user-account-edit-permissions.component.ts (hunk 2 above)
Comment (ChazUK) [reply src/app/features/workplace/user-account-edit-permissions/user-account-edit-permissions.component.ts]: agreed, will update

File: src/app/features/workplace/user-account-edit-permissions/user-account-edit-permissions.component.ts (hunk 1 above)
Comment (ChazUK) [reply src/app/features/workplace/user-account-edit-permissions/user-account-edit-permissions.component.ts]: Explained on Slack that only the "Return to user accounts" button needs to navigate back.

File: src/app/features/workplace/user-account-edit-permissions/user-account-edit-permissions.component.ts (hunk 2 above)
Comment (ChazUK) [reply src/app/features/workplace/user-account-edit-permissions/user-account-edit-permissions.component.ts]: same as above

**Pull request merged**